config/routes.rb

```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
   root to: "wishlists#index"
-  resources :wishlists
+  resources :wishlists do
+    resources :items
```

Comment:
You've only added `#new` and `#create` actions in the controller. And `#show` which is not needed. Restrict the routes to those actions only.